### PR TITLE
Extinct Animals Pack by Helixen

### PR DESCRIPTION
**Pull request by Breadbox2k19**
## Additions
Everything patched. Something simple this time.

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors
- [x] (For compatibility patches) ...with and without patched mod loaded
- [x] Playtested a colony (specify how long)

### CODE REVIEW

File: Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+							<cooldownTime>2.33</cooldownTime>
+							<linkedBodyPartsGroup>HornAttackTool</linkedBodyPartsGroup>
+							<armorPenetrationBlunt>22</armorPenetrationBlunt>
+						</li>
+						<li Class="CombatExtended.ToolCE">
+							<label>horn</label>
+							<capacities><li>Stab</li><li>Scratch</li></capacities>
+							<power>36</power>
+							<cooldownTime>2.33</cooldownTime>
+							<linkedBodyPartsGroup>HornAttackTool</linkedBodyPartsGroup>
```
Comment (N7Huntsman) [inline Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:135]: Reference the existing horn attacks for rhinos. I don't know what, if any, impact reusing the same linkedBodyPartsGroup has.
```
				<li Class="CombatExtended.ToolCE">
					<label>horn</label>
					<capacities>
						<li>Poke</li>
					</capacities>
					<power>24</power>
					<cooldownTime>3.17</cooldownTime>
					<linkedBodyPartsGroup>HornAttackTool</linkedBodyPartsGroup>
					<armorPenetrationBlunt>10</armorPenetrationBlunt>
				</li>
				<li Class="CombatExtended.ToolCE">
					<label>horn</label>
					<capacities>
						<li>Blunt</li>
					</capacities>
					<power>24</power>
					<cooldownTime>3.17</cooldownTime>
					<linkedBodyPartsGroup>HornAttackTool_2</linkedBodyPartsGroup>
					<armorPenetrationBlunt>10</armorPenetrationBlunt>
				</li>
				```

File: Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+							<capacities><li>Blunt</li></capacities>
+							<power>30</power>
+							<cooldownTime>2.33</cooldownTime>
+							<linkedBodyPartsGroup>HornAttackTool</linkedBodyPartsGroup>
+							<armorPenetrationBlunt>22</armorPenetrationBlunt>
+						</li>
+						<li Class="CombatExtended.ToolCE">
+							<label>horn</label>
+							<capacities><li>Stab</li><li>Scratch</li></capacities>
+							<power>36</power>
```
Comment (N7Huntsman) [inline Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:133]: I'd scale this down and remove the `Scratch` capability, and may consider swapping the Stab for a Poke to match the vanilla rhino. The vanilla rhino does 24 damage, and while the wooly rhino is slightly larger from what I can tell, it's not much larger. I'd set both horn attacks to 28 - 32 range.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Generally looks good, a few of the animals need to be scaled back just a bit.

File: Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+				<value>
+					<li Class="CombatExtended.RacePropertiesExtensionCE">
+						<bodyShape>Quadruped</bodyShape>
+					</li>
+				</value>
+			</li>
+
+			<li Class="PatchOperationAdd">
+				<xpath>/Defs/ThingDef[defName="RG-EAP-Protoceratidae"]/statBases</xpath>
+				<value>
```
[the remark this answers is not in the document]
Comment (Breadbox2k19) [reply Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:22]: I'll make them faster, I kinda just randomly set the values for parry.

File: Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml (hunk 1 above)
Comment (Breadbox2k19) [reply Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:135]: Rhino attacks hit harder after the recent vanilla animal tweak. So I'm basing the stats of the woolly rhino on that.

**Review by N7Huntsman — APPROVED** [review]

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
Extinct_Mammals patch is currently failing. I'm not certain which op.

File: Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+							<cooldownTime>2.7</cooldownTime>
+							<linkedBodyPartsGroup>HeadAttackTool</linkedBodyPartsGroup>
+							<armorPenetrationBlunt>24</armorPenetrationBlunt>
+						</li>
+					</tools>
+				</value>
+			</li>
+
+        <li Class="PatchOperationReplace">
+		<xpath>Defs/PawnKindDef[defName="RG-EAP_WoollyRhino"]/race/baseHealthScale</xpath>
```
Comment (N7Huntsman) [inline Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:154]: This appears to be the patch op that is broken.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
- Broken Woolly Rhino operation.
- Please remove (blunt) from the attack labels, as it'll already show the damage type.

File: Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml (hunk 4 above)
Comment (Breadbox2k19) [reply Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:154]: Should all be fixed now.

Comment (N7Huntsman) [reply Patches/HLX ReGrowth - Extinct Animals Pack/Extinct_Mammals.xml:154]: It's not, but I figured out what's wrong--it should be a `ThingDef` not a `PawnKindDef`.

**Review by N7Huntsman — CHANGES_REQUESTED** [review]
One op stilling failing, but I found the issue.

**Review by N7Huntsman — APPROVED** [review]

**Pull request merged**